src/common/buffer.cc

```diff
@@ -1768,7 +1768,7 @@ static std::atomic_flag buffer_debug_lock = ATOMIC_FLAG_INIT;
   {
     if (append_buffer.unused_tail_length() < prealloc) {
       append_buffer = buffer::create(prealloc);
-      if (_mempool) {
+      if (_mempool >= 0) {
```

Comment:
yeah, it's `-1` by default. and the mem pool index starts at 0. they are enum whose first element is not assigned with a number.